api/src/main/java/org/openmrs/module/registrationcore/api/impl/PatientUpdatedListener.java

```diff
@@ -0,0 +1,69 @@
+package org.openmrs.module.registrationcore.api.impl;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.openmrs.Patient;
+import org.openmrs.event.Event;
+
+import javax.jms.Message;
+import org.openmrs.module.registrationcore.api.errorhandling.ErrorHandlingService;
+import org.openmrs.module.registrationcore.api.errorhandling.PixErrorHandlingService;
+import org.openmrs.module.registrationcore.api.mpi.common.MpiException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class listens for patient UPDATED events.
+ * If MPI is enabled it updates patient in MPI.
+ */
+public class PatientUpdatedListener extends PatientActionListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PatientUpdatedListener.class);
+
+    /**
+     * Defines the list of Actions that this subscribable event listener class listens out for.
+     *
+     * @return a list of Actions this listener can deal with
+     */
+    public List<String> subscribeToActions(){
+        List actions = new ArrayList<String>();
+        actions.add(Event.Action.UPDATED.name());
+        return actions;
+    }
+
+    /**
+     * Update patient in MPI server.
+     *
+     * @param message message with properties.
+     */
+    // TODO what should we do if patient is voided? discuss
+    @Override
+    public void performMpiAction(Message message) {
+        Patient patient = extractPatient(message);
+        // if patient voided is true then don't update the MPI
+        if (patient.getVoided()){
+            return;
+        }
+        else {
```

Comment:
else is redundant here because you return in the if clause

Reply:
removed